### ISSUE TITLE
Double inheritance of enum.Enum -> false __members__ is unsubscriptable

### ISSUE THREAD

**Issue by mhooreman** [PyCQA/pylint#3535]
Hello,

When we inherit of a class which itself inherits of `enum.Enum`, the members are detected as unsubscriptable.

Note: I use this because I define a "enum with additional features".

### Steps to reproduce

```python
import enum

class A(enum.Enum):
    pass

class B(A):
    x = enum.auto()

print(B.__members__['x'])
```
### Current behavior

`Value 'B.__members__' is unsubscriptable`

### Expected behavior

No such issue

### pylint --version output

```
pylint 2.5.0
astroid 2.4.0
Python 3.7.4 (v3.7.4:e09359112e, Jul  8 2019, 14:54:52) 
[Clang 6.0 (clang-600.0.57)]
```

**Comment by tillhainbach** [comment]
Also throws an error if one try's to use the .keys()-method. (Iterating over names and aliases). Additionally, throws an error if one creates "mixed-in" enums (e.g. as mentioned in the official python docs: https://docs.python.org/3/library/enum.html#others).

Example:
```python
"""Just an example"""

from enum import Enum

class StrEnum(str, Enum):
    """Self-declared mixed-in enum (like IntEnum)"""

class MyEnum(StrEnum):
    """An Enum with aliases"""
    my_value = 1
    my_other_value = 2
    my_alias = 1

def filter_names_in_my_enum(names):
    """Get an overview of all names in Enum with aliases."""
    names = (name for name in names if name in MyEnum.__members__.keys())
    return names

def main():
    """Main method"""
    names = ['my_value', 'my_int', 'my_alias']
    print(list(filter_names_in_my_enum(names)))


if __name__ == '__main__':
    main()
```

```bash
% pylint example.py
example.py:16:47: E1101: Function '__members__' has no 'keys' member (no-member)
------------------------------------------------------------------
Your code has been rated at 6.43/10
% python example.py
['my_value', 'my_alias']
```
For some reason, this works:
```python
"""Just an example2"""

from enum import Enum

class StrEnum(str, Enum):
    """Self-declared mixed-in enum (like IntEnum)"""

class MyEnum(StrEnum):
    """An Enum with aliases"""
    my_value = 1
    my_other_value = 2
    my_alias = 1

def filter_names_in_the_enum(names, the_enum):
    """Get an overview of all names in Enum with aliases."""
    # call .__members__.keys() on function argument
    names = (name for name in names if name in the_enum.__members__.keys())
    return names

def main():
    """Main method"""
    names = ['my_value', 'my_int', 'my_alias']
    # pass MyEnum as argument to function
    print(list(filter_names_in_the_enum(names, MyEnum)))


if __name__ == '__main__':
    main()
```

```bash
% pylint example2.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 

% python example2.py
['my_value', 'my_alias']
```
